test/v3/connectors/RetrieveBalanceAndTransactionsConnectorSpec.scala

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v3.connectors
+
+import api.config.MockAppConfig
+import api.connectors.{ConnectorSpec, DownstreamOutcome, MockHttpClient}
+import api.models.domain.{DateRange, Nino}
+import api.models.outcomes.ResponseWrapper
+import v3.fixtures.retrieveBalanceAndTransactions.BalanceDetailsFixture.balanceDetails
+import v3.fixtures.retrieveBalanceAndTransactions.CodingDetailsFixture.codingDetails
+import v3.fixtures.retrieveBalanceAndTransactions.DocumentDetailsFixture.{documentDetails, documentDetailsWithoutDocDueDate}
+import v3.fixtures.retrieveBalanceAndTransactions.FinancialDetailsFixture.financialDetailsFull
+import v3.models.request.retrieveBalanceAndTransactions.RetrieveBalanceAndTransactionsRequestData
+import v3.models.response.retrieveBalanceAndTransactions._
+
+import java.time.LocalDate
+import scala.concurrent.Future
+
+class RetrieveBalanceAndTransactionsConnectorSpec extends ConnectorSpec {
+
+  private val nino                       = "AA123456A"
+  private val docNumber                  = "anId"
+  private val fromDate                   = "2018-08-13"
+  private val toDate                     = "2019-08-13"
+  private val onlyOpenItems              = false
+  private val includeLocks               = false
+  private val calculateAccruedInterest   = false
+  private val removePOA                  = false
+  private val customerPaymentInformation = false
+  private val includeEstimatedCharges    = false
+
+  private val validResponse: RetrieveBalanceAndTransactionsResponse =
+    RetrieveBalanceAndTransactionsResponse(
+      balanceDetails = balanceDetails,
+      codingDetails = Some(List(codingDetails)),
+      documentDetails = Some(List(documentDetails, documentDetailsWithoutDocDueDate)),
+      financialDetails = Some(List(financialDetailsFull))
+    )
+
+  private val validRequest: RetrieveBalanceAndTransactionsRequestData = RetrieveBalanceAndTransactionsRequestData(
+    nino = Nino(nino),
+    docNumber = Some(docNumber),
+    Some(DateRange(LocalDate.parse(fromDate), LocalDate.parse(toDate))),
+    onlyOpenItems = onlyOpenItems,
+    includeLocks = includeLocks,
+    calculateAccruedInterest = calculateAccruedInterest,
+    removePOA = removePOA,
+    customerPaymentInformation = customerPaymentInformation,
+    includeEstimatedCharges = includeEstimatedCharges
+  )
```

Comment:
When the arg types are just booleans or strings rather than "proper" types, it guards against args being swapped around somewhere. I've removed them anyway so it's a bit tidier